### PR TITLE
FUSETOOLS-2316

### PR DESCRIPTION
**Pull request by apupier**
- Provide Remote Camel Debug (without java because the java can be not
enabled and only JMX is available)
- Provide Remote Camel and Java debug
- Clean code
- Provide disconnection only for remote process to be coherent with Java
Debugger

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.launcher.ui/plugin.xml [hunk 1]
```diff
@@ -18,6 +28,17 @@
             id="org.fusesource.ide.launcher.ui.launchConfigurationTabGroup.camelContext"
             type="org.fusesource.ide.launcher.camelContext">
       </launchConfigurationTabGroup>
+      <launchConfigurationTabGroup
+            class="org.fusesource.ide.launcher.ui.remote.tabs.RemoteCamelContextAndJavaLaunchConfigurationTabGroup"
+            description="Update settings to connect to a remote Camel Context and Java vm"
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher.ui/plugin.xml]: please use i10n

File: editor/plugins/org.fusesource.ide.launcher/plugin.xml [hunk 2]
```diff
@@ -13,6 +13,21 @@
             sourceLocatorId="org.fusesource.ide.launcher.debug.sourceLookupDirector.camel"
             sourcePathComputerId="org.fusesource.ide.launcher.debug.sourcePathComputer.camel">
       </launchConfigurationType>
+      <launchConfigurationType
+            delegate="org.fusesource.ide.launcher.remote.debug.RemoteCamelAndJavaLaunchConfigurationDelegate"
+            id="org.fusesource.ide.launcher.camelandjavacontext.remote"
+            modes="debug"
+            name="Remote Camel and Java Context"
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher/plugin.xml]: please use i10n

File: editor/plugins/org.fusesource.ide.launcher/plugin.xml [hunk 3]
```diff
@@ -13,6 +13,21 @@
...
+            sourceLocatorId="org.fusesource.ide.launcher.debug.sourceLookupDirector.camel"
+            sourcePathComputerId="org.fusesource.ide.launcher.debug.sourcePathComputer.camel">
+      </launchConfigurationType>
+      <launchConfigurationType
+            delegate="org.fusesource.ide.launcher.remote.debug.RemoteCamelLaunchConfigurationDelegate"
+            id="org.fusesource.ide.launcher.camelcontext.remote"
+            modes="debug"
+            name="Remote Camel Context"
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher/plugin.xml]: please use i10n

File: editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/model/CamelDebugTarget.java [hunk 4]
```diff
@@ -256,44 +256,35 @@ private boolean connectToVM() {
 			return true;
 		} catch (IOException ex) {
 			//ignore
+			System.out.println(ex);
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/model/CamelDebugTarget.java]: use a logger

File: pom.xml [hunk 5]
```diff
@@ -437,7 +437,8 @@
 				<sonar.code.codeCoveragePlugin>jacoco</sonar.code.codeCoveragePlugin>
 				<sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
 				<!-- points to the target folder of the global parent project -->
-				<sonar.jacoco.reportPath>../../target/jacoco.exec</sonar.jacoco.reportPath>
+				<!-- <sonar.jacoco.reportPath>../../target/jacoco.exec</sonar.jacoco.reportPath> -->
```
Comment (lhein) [inline pom.xml]: is that already in sync with your Sonar changes?

Comment (apupier) [inline pom.xml]: no, will need to rebase and resolve conflict

**Pull request merged**